### ISSUE TITLE
Invalid JWT: Token must be a short-lived token (60 minutes) and in a reasonable timeframe. Check your iat and exp values in the JWT claim

### ISSUE THREAD

**Issue by afloarea** [eclipse-vertx/vertx-auth#583]
### Version

4.3.3

### Context

I was trying to connect to google spreadsheets API using a service account. After some trial and error, I figured I can use something of the sorts:
```
final var auth = GoogleAuth.create(vertx, serviceAccountConfig); // service account config file downloaded from google
final var client = WebClient.create(vertx);

final var oath2Client = OAuth2WebClient.create(client, auth)
	.withCredentials(new Oauth2Credentials()
		.setFlow(OAuth2FlowType.AUTH_JWT)
		.setJwt(new JsonObject().put("scope", String.join(" ", List.of("https://www.googleapis.com/auth/spreadsheets","https://www.googleapis.com/auth/drive")))));
```

At first, it seems to work and it looks pretty cool and simple, however, after about an hour I start to get an error:
`Invalid JWT: Token must be a short-lived token (60 minutes) and in a reasonable timeframe. Check your iat and exp values in the JWT claim after one hour`

Trying to investigate, the internet says that this is probably caused by the clock being out of sync (NTP). However this is was not the case. After more digging I found out that after expiration, the same flow (JWT_AUTH) was being performed again but it failed because of an invalid `iat` field. It turns out that the JsonObject passed with `.setJwt(...)` was altered and new fields were added by the JWT::sign(JsonObject,JWTOptions) method (which set the `iat` field on the first call to google API). After expiration, the sign method runs again, only this time it finds that the payload already has an `iat` field and so, it leaves it as is, which google does not like and returns the before-mentioned error.

### Do you have a reproducer?

Unfortunately I do not, as it involves credentials.

### Steps to reproduce

1. obtain a pair of credentials for service account in the form:
```
{
	"type": "service_account",
	"project_id": "<proj-id>",
	"private_key_id": "<id>",
	"private_key": "<private_key_cert>",
	"client_email": "email@test.com",
	"client_id": "<client_id_value>",
	"auth_uri": "https://accounts.google.com/o/oauth2/auth",
	"token_uri": "https://oauth2.googleapis.com/token",
	"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
	"client_x509_cert_url": "<cert_url_value>"
}
```

2.  Create the web client as mentioned above
4.  Make a request to some google api
5.  Wait a bit over an hour (70 mins)
6. Try to make the same request again and it should fail with the above-mentioned error

### Extra

As a workaround, I tried setting a periodic timer that runs every 30 mins and calls the 'withCredentials' method on the OAuth2WebClient and I did no longer receive the error.

I think that making a copy of the payload in the JWT::sign(JsonObject,JWTOptions) method at the start of the method will probably solve the issue, but there might be a better way (maybe not altering the Credentials object passed by the Vert.x toolkit user)

**Comment by pmlopes** [comment]
Indeed we shouldn't modify the user supplied state.

**Comment by pendula95** [comment]
Fixed by #585
